### ISSUE TITLE
Switching between traditional and simplified

### ISSUE THREAD

**Issue by chrix** [batterseapower/pinyin-toolkit#148]
Hi,

I mostly use traditional, so it's a great feature to have the simplified displayed as well.
Now I have some sets with simplified as the basis, where I'd like to have the traditional displayed without changing the global settings for pinyin toolkit.

Any way to do this? Creating a "traditional" field and running the "fill missing card data" routine didn't seem to help...

Thanks

**Comment by batterseapower** [comment]
After running that routine, the Traditional fields on all your cards should be filled. You should be able to check that using the fact browser.

After you've done that, you'll need to modify the templates for your cards so that they display the Traditional field alongside the Expression (which will be simplified, as you entered it).

If the "fill missing data" didn't fill out the traditional field at all, then thats a bug with the toolkit and we need to fix it. Otherwise, you just need to make that change to the template. Let me know if you have trouble with it.

**Comment by chrix** [comment]
Yeah, I changed the template and also checked it using the fact browser, to no avail... My other decks (with traditional as the basis) have simplified subfields that were automatically created by pinyin toolkit, so on that end it's all working great, just when it's going the other way there seem to be some problems...

**Comment by batterseapower** [comment]
Sorry, it's not yet clear to me what is going wrong. Do you mean that the "Traditional" field for your facts hasn't been filled out by the Toolkit?

**Comment by chrix** [comment]
Yes. My global settings for the pinyin toolkit are set to traditional since that is what I mostly use (so in most of my decks, the "expression" field is in traditional). Just one deck or two are in simplified (i.e. the "expression" field is in simplified), for which I'd like to display the traditional script in the "traditional" field. And that doesn't work

**Comment by batterseapower** [comment]
I'm not sure you've understood my question, or perhaps I haven't understood your answer. Let me rephrase:

The deck stores a number of "facts". A fact has a number of "fields". In your deck one of these "fields" is called "Traditional". When you ran "fill missing data" that field should have been filled out (in all your facts!) with the traditional version of the "Expression" field.

What do you see if you go to the fact browser, choose one of those facts, and look at the field list at the bottom of the window? Does the "Traditional" field on the fact indeed contain the traditional version of the expression or not?

Note that this is NOT a question about what you see on the "cards", which are generated from the facts. I'm trying to work out if your problem is with the cards (templates) or with your facts.

**Comment by chrix** [comment]
oh yes, that's how I understood your first question, and that's what I did, checking the "Traditional" field (which I created prior to running the routine) in the fact browser. There's nothing in it.

**Comment by batterseapower** [comment]
OK, thanks for the clarification! Just a couple of quick checks:
1) Is the checkbox next to the Traditional field ticked in the preferences dialog?
2) What happens if go to a fact in your deck without anything in the traditional field and tab away from the expression? Dose the traditional field get filled?
3) What happens if you add a new fact containing something like 个 and tab away from the expression? Do you get the traditional in the traditional field as expected?

Note that IIRC the traditional field will not be filled if the traditional doesn't differ from the expression.

**Comment by chrix** [comment]
1) it's not. Whenever I try to check it, an error occurs. (how do you quote code on this platform, I would like to cut and paste the error message)
If I keep it checked, then I can no longer access the properties of the toolkit, with the same error occurring. Luckily I found a way to get around this (somewhere on this site),  delete config.db in the .anki directory. 

2) & 3): no.

**Comment by chrix** [comment]
Sorry, just to add: so neither the Simplified nor the Traditional fields are ticked, but nevertheless the automatic filling mechanism works for Traditional -> Simplified.

**Comment by batterseapower** [comment]
1) You can quote code by indenting it by 4 spaces or 1 tab. I'd be very interested in seeing the error. This sounds like a serious problem - but I've just checked and it seems to work OK for me in the latest release.

This _might_ have been caused by some settings in config.db from a previous version. I've tried quite hard to make this work correctly but it's hard to handle all cases. If this is the case, then deleting the config.db should have cleared the problem and allow you to check the "simplified" character setting without an error.

I'm really puzzled about the Simplified being filled despite the checkbox being unticked! Perhaps that error will reveal the cause.

(By the way, I assume you are using the latest release - 0.6.1?)

Sorry that the toolkit isn't working for you!

**Comment by chrix** [comment]
the error does re-occur
    Traceback (most recent call last):
    File             "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\forms\preferencescontroller.py", line 316, in 
    self.checkbox.connect(self.checkbox, SIGNAL("clicked()"), lambda: self.updateModel())
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\forms\preferencescontroller.py", line 319, in updateModel
    self.updateModelValue(self.checkbox.isChecked())
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\forms\preferencescontroller.py", line 290, in updateModelValue
    self.modelchanged.fire()
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\forms\preferencescontroller.py", line 276, in fire
    subscriber(_args, *_kwargs)
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\forms\preferencescontroller.py", line 228, in updateViewPreview
    self.updaterfromexpr.updatefact(fact, previewexpressions[self.model.prefersimptrad])
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\updater.py", line 374, in updatefact
    value = updater()
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\updater.py", line 346, in 
    'simp'       : lambda: (expressionviews["trad"] != expressionviews["simp"]) and expressionviews["simp"] or None,
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\utils.py", line 494, in **getitem**
    value = self.factory(key)
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\updater.py", line 326, in 
    expressionviews = utils.FactoryDict(lambda simptrad: self.generateincharactersystem(expression, simptrad))
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\updater.py", line 222, in generateincharactersystem
    meanings = dictionaryonline.gTrans(expression, glangcode, False)
  File "C:\Users\Chris\AppData\Roaming.anki\plugins\pinyin\dictionaryonline.py", line 40, in gTrans
    log.exception("Error while parsing Google response: %s" % repr(literal))
NameError: global name 'literal' is not defined

**Comment by chrix** [comment]
sorry somehow the quoting didn't work out the way I thought it would... I'm not sure where I can find the version, but I just re-installed anki two days ago due to the same error occurring (but unrelated to this problem)

But I just wanna say, the toolkit is working for me, with all of my traditional-based decks anyways, I really appreciate all that effort you guys put into this!

**Comment by batterseapower** [comment]
OK, I see what the problem is now. You need to upgrade to the latest version (0.6.1) which I released yesterday. It fixes that exact error - it is caused by a change to the format Google returns translations in. After you have done that, ticking the checkboxes will work and it should behave as you expect.

Let me know if you have trouble after the upgrade.

**Comment by chrix** [comment]
thanks, it works now!

**Comment by batterseapower** [comment]
Hurrah! I'm glad we have got this sorted out :-)
